### PR TITLE
feat: add python `sys.path` to vyper path

### PR DESCRIPTION
**Pull request by charles-cooper**
this allows users to install packages from pip and import them using a regular python workflow.

### What I did
opening pandora's box

### How I did it
inject `sys.path` into the vyper compiler's search path

### How to verify it
```bash
pip install snekmate
cat > test.vy <<EOF
from auth import AccessControl as acl  # AccessControl.vy is in snekmate
EOF
vyper test.vy
```

### Commit message

```
this makes it easier to install vyper packages from pip and import them
using a regular python workflow.

misc:
- improve how paths appear in error messages; try hard to make them
  relative paths.
- add `chdir_tmp_path` fixture which chdirs to the `tmp_path` fixture
  for the duration of the test.
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by fubuloubu — CHANGES_REQUESTED** [review]
Can the package name be used as the base of the namespace?

e.g. `from snekmate.auth import ...`

**Comment by charles-cooper** [comment]
> Can the package name be used as the base of the namespace?
> 
> e.g. `from snekmate.auth import ...`

that's an issue for the package maintainer -- snekmate is currently installing `auth` to the top- level of `site-packages/`. better hygiene would be for it to be installed to `site-packages/snekmate/`, then it would be imported as `from snekmate.auth import AccessControl`.

**Comment by pcaversaccio** [comment]
> > Can the package name be used as the base of the namespace?
> > e.g. `from snekmate.auth import ...`
> 
> that's an issue for the package maintainer -- snekmate is currently installing `auth` to the top- level of `site-packages/`. better hygiene would be for it to be installed to `site-packages/snekmate/` (cc. @pcaversaccio ), then it would be imported as `from snekmate.auth import AccessControl`.

This is correct - I opened a PR to fix this pattern in snekmate: https://github.com/pcaversaccio/snekmate/pull/204 (@charles-cooper can you quickly have a look). If you use the fixed pattern via:

```console
pip install -i https://test.pypi.org/simple/ snekmate==0.0.5rc2
```

and the introduced changes in this PR, you will get a different behaviour for Linux vs Windows for the following test file:

```vy
# Test.vy
from snekmate.auth import AccessControl as acl
```

## Ubuntu

```console
Error compiling: Test.vy
vyper.exceptions.ModuleNotFound: vyper.interfaces.ERC165
  contract "/mnt/c/Users/pasca/Desktop/vyper/venvU/lib/python3.11/site-packages/snekmate/auth/AccessControl.vy:54", line 54:0
       53 # which is a built-in interface of the Vyper compiler.
  ---> 54 from vyper.interfaces import ERC165
  --------^
       55 implements: ERC165
```

## Windows

```console
Error compiling: Test.vy
vyper.exceptions.ModuleNotFound: snekmate.auth.AccessControl

  contract "Test.vy:1", line 1:0
  ---> 1 from snekmate.auth import AccessControl as acl
  -------^
       2
```

**TL;DR:** The compilation error currently on Ubuntu is due to the recent change introduced here https://github.com/vyperlang/vyper/pull/3741 (I would need to refactor this to `from ethereum.ercs import ERC165` in the next release), but it works. On Windows it doesn't seem to work tho - am not sure why 🤔.

**Review by pcaversaccio — COMMENTED** [review]
Can we add one test case please?

**Comment by charles-cooper** [comment]
> If I adjust the snekmate AccessControl.vy module locally to use from ethereum.ercs import ERC165, I however get the following raise:

seems like an outstanding issue

**Review by harkal — APPROVED** [review]

**Review by fubuloubu — COMMENTED** [review]
Figure out a way to mock this for testing

**Comment by charles-cooper** [comment]
test added here https://github.com/vyperlang/vyper/pull/3763/commits/eac2d2f1fde91fda27e48d99e8ddcee274a86c1e

**Review by fubuloubu — APPROVED** [review]

**Comment by charles-cooper** [comment]
> > If I adjust the snekmate AccessControl.vy module locally to use from ethereum.ercs import ERC165, I however get the following raise:
> 
> seems like an outstanding issue

error message should be fixed in https://github.com/vyperlang/vyper/pull/3775
